data/templates/template.typst

```diff
@@ -29,9 +29,13 @@
 ) = {
   set document(
     title: title,
+  ) if title != none
+  set document(
     author: authors.map(author => content-to-string(author.name)),
+  ) if authors != none
+  set document(
     keywords: keywords,
-  )
+  ) if keywords != none
```

Comment:
If I'm not mistaken, `authors` is the only one that really needs this treatment.  For the others, we have checks for `!= none` down below where they are used.  Why move the `content-to-string` business into the clause below where `authors` are rendered -- it already has a check for `none` -- and then this part can be simple again?

Reply:
The error I see in the issue is:
`expected string or array, found none`.

This fixes that because `content-to-string` breaks on `none` since you can't access fields.

Is there another issue?

Reply:
Oh, I understand now; I wasn't looking at the whole context.

I reverted your original change.  Did you want to integrate those back into this PR?  OR should I apply this now first?

Reply:
Conceptually, yes, `content-to-string` creates strings, but it's operating on the map, which is creating an array of strings. This block converts it from an array of `content` which is Typst's type for (broadly put) structured text to an array of `str`s. Then that array is joined element-wise with `,` or `&` if it's the last entry.

Let me rebuild this sometime today and verify.